src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/dashboard-area-chart/dashboard-area-chart.component.ts

```diff
@@ -176,28 +177,57 @@ export class DashboardAreaChartComponent implements OnInit, OnChanges, AfterView
         this.data2[this.data2.length - 1][1]
       ).split(' ');
     }
-  }
-
-  ngAfterViewInit(): void {
-    if (this.data) {
-      this.setChartTicks();
+    if (this.chart) {
+      this.chart.chart.update();
     }
   }
 
   private formatData(array: Array<any>): any {
     let formattedData = {};
     formattedData = array.map((data: any) => ({
       x: data[0] * 1000,
-      y: Number(this.convertUnits(data[1]).replace(/[^\d,.]+/g, ''))
+      y: Number(this.convertToChartDataUnits(data[1]).replace(/[^\d,.]+/g, ''))
     }));
     return formattedData;
   }
 
+  private convertToChartDataUnits(data: any): any {
+    let dataWithUnits: string = '';
+    if (this.chartDataUnits) {
+      if (this.dataUnits === 'B') {
+        dataWithUnits = this.numberFormatter.formatBytesFromTo(
+          data,
+          this.dataUnits,
+          this.chartDataUnits
+        );
+      } else if (this.dataUnits === 'B/s') {
+        dataWithUnits = this.numberFormatter.formatBytesPerSecondFromTo(
+          data,
+          this.dataUnits,
+          this.chartDataUnits
+        );
+      } else if (this.dataUnits === 'ms') {
+        dataWithUnits = this.numberFormatter.formatSecondsFromTo(
+          data,
+          this.dataUnits,
+          this.chartDataUnits
+        );
+      } else {
+        dataWithUnits = this.numberFormatter.formatUnitlessFromTo(
+          data,
+          this.dataUnits,
+          this.chartDataUnits
+        );
+      }
```

Comment:
No need to repeat everything, you're just looking for the proper method based on the unit type, so a switch-case or an associate map would do the trick:
```suggestion
      switch (this.dataUnits) {
        case 'B':
          label = this.bytesLabels;
          break;
        case 'B/s':
          ...
      }
      dataWithUnits = this.formatFromTo(
        data,
        this.dataUnits,
        this.chartDataUnits,
        label
      );
```